.golangci.yaml

```diff
@@ -1,24 +1,138 @@
-linters-settings:
-  misspell:
-    ignore-words:
-    - creater
+run:
+  timeout: 10m
+  allow-parallel-runners: true
 
 linters:
+  disable-all: true
   enable:
+  - asasalint
   - asciicheck
   - bidichk
+  - bodyclose
+  - containedctx
+  - depguard
+  - dupword
   - durationcheck
-  - errname
-  - errorlint
+  - errcheck
+  - errchkjson
   - exportloopref
+  - gocritic
+  - godot
   - gofmt
+  - goprintffuncname
+  - gosec
   - gosimple
   - govet
   - importas
   - ineffassign
   - misspell
   - nilerr
+  - noctx
+  - nolintlint
+  - nosprintfhostport
+  - prealloc
+  - revive
+  - staticcheck
+  - thelper
   - unconvert
   - unused
-  - errcheck
-  - nolintlint
+  - usestdlibvars
+  - whitespace
+  # TODO(vincepri): Figure out if we want to enable or remove the following linters:
+  # - predeclared
+  # - goconst
+
+linters-settings:
+  misspell:
+    ignore-words:
+    - creater
+  goconst:
+    ignore-tests: true
+  nolintlint:
+    allow-unused: false
+    allow-leading-space: false
+    require-specific: true
+  revive:
+    revive:
+    rules:
+      - name: context-keys-type
+      - name: error-return
+      - name: error-strings
+      - name: error-naming
+      - name: if-return
+      - name: increment-decrement
+      - name: var-declaration
+      - name: package-comments
+      - name: range
+      - name: receiver-naming
+      - name: time-naming
+      - name: errorf
+      - name: superfluous-else
+      - name: unreachable-code
+      - name: bool-literal-in-expr
+      - name: constant-logical-expr
+      # TODO(vincepri): Figure out if we want to enable
+      # the following rules, or remove them completely, they're a bit noisy.
+      # - name: context-as-argument
+      # - name: var-naming
+      # - name: exported
+      # - name: unexported-return
+      # - name: blank-imports
+      # - name: indent-error-flow # I think @ncdc prefers explicit else statements, remove?
+      # - name: redefines-builtin-id
+      # - name: dot-imports
+  gosec:
+    excludes:
+    - G307 # Deferring unsafe method "Close" on type "\*os.File"
+    - G108 # Profiling endpoint is automatically exposed on /debug/pprof
+    # TODO(vincepri): The following should be looked at and removed in future iterations.
```

Comment:
same question as above, do we have a tracking issue for this one?

Reply:
Opened https://github.com/kcp-dev/kcp/issues/2539